library/std/src/sys/thread_local/mod.rs

```diff
@@ -135,6 +137,7 @@ pub(crate) mod key {
                 target_family = "unix",
             ),
             target_os = "teeos",
+            all(target_os = "wasi", target_env = "p1", target_feature = "atomics"),
```

Comment:
When users compile to wasm32-wasip1-threads, they don't necessarily enable `target_feature = "atomics"`, but the rust-std of `wasm32-wasip1-threads` always enables atomics, which causes some issues: https://github.com/rust-lang/rust/issues/137510.

Reply:
Move this to https://github.com/rust-lang/rust/issues/137510#issuecomment-2677587008